include/deal.II/gmsh/utilities.h

```diff
@@ -94,6 +104,60 @@ namespace Gmsh
     Triangulation<2, spacedim> &tria,
     const AdditionalParameters &prm = AdditionalParameters());
 #  endif
+
+#  ifdef DEAL_II_GMSH_WITH_API
+  /**
+   * Read a partitioned GMSH mesh file and create a fully distributed
+   * triangulation in deal.II.
+   *
+   * This function reads partitioned mesh files generated by GMSH's internal
+   * partitioner (usually metis) and constructs a fully distributed
+   * triangulation that can be used in parallel computations with MPI. The
+   * function automatically handles ghost cell identification, vertex
+   * coordinates, and cell connectivity across processor boundaries.
+   *
+   * In order to generate a set of partitioned mesh files that can be used with
+   * this function, you could use the `GMSH` command line tool with the
+   * following options:
+   * @code
+   * gmsh -part 4 -part_ghosts -part_split -save {input_file}
+   * @endcode
+   *
+   * This will generate a set of mesh files named like `{input_file}_1.msh`,
+   * `{input_file}_2.msh`, etc.
+   *
+   * In the command above, {input_file} can be any valid GMSH mesh input file.
+   *
+   * The mesh files are expected to follow the naming convention:
+   * - For single processor: `{file_prefix}.{file_suffix}`
+   * - For multiple processors: `{file_prefix}_{rank+1}.{file_suffix}` where
+   *   rank goes from 1 to number of processors
+   *
+   * @tparam dim       The dimension of the triangulation (1, 2, or 3).
+   * @tparam spacedim  The spatial dimension in which the triangulation lives.
+   *
+   * @param[out] tria         The distributed triangulation to be populated.
+   * @param[in]  mpi_comm     The MPI communicator.
+   * @param[in]  file_prefix  Prefix of mesh files.
+   * @param[in]  file_suffix  Suffix of mesh files.
+   *
+   * @pre  Triangulation must be empty.
+   * @pre  Mesh files must match the number of MPI ranks. An exception is thrown
+   * if there are more (or less) files than ranks.
+   *
+   * @note Mesh file is partitioned by Gmsh api using command
+   * @note The function uses GMSH's API to read the mesh and requires GMSH to be
+   * initialized before calling this function.
+   */
+  template <int dim, int spacedim>
+  void
+  read_partitioned_msh(
+    parallel::fullydistributed::Triangulation<dim, spacedim> &tria,
+    const MPI_Comm                                            mpi_comm,
```

Comment:
Couldn't you get the communicator out of the `tria`? That would make sure that there isn't going to be any mismatch in the arguments.